src/services/sessions.js

```diff
@@ -750,13 +882,29 @@ module.exports = class SessionsHelper {
 	 * @returns {JSON} - Enroll session.
 	 */
 
-	static async enroll(sessionId, userTokenData, timeZone) {
-		const userDetails = (await userRequests.details('', userTokenData.id)).data.result
-
-		const userId = userTokenData.id
-		const email = userDetails.email
-		const name = userDetails.name
+	static async enroll(sessionId, userTokenData, timeZone, isSelfEnrolled = true) {
 		try {
+			let email
+			let name
+			let userId
+			let enrollmentType
+			let emailTemplateCode = process.env.MENTEE_SESSION_ENROLLMENT_EMAIL_TEMPLATE
+			// If entrolled by the mentee get email and name from user service via api call.
+			// Else it will be avalable in userTokenData
+			if (isSelfEnrolled) {
+				const userDetails = (await userRequests.details('', userTokenData.id)).data.result
+				console.log('timeZone : ', timeZone)
+				userId = userDetails.id
+				email = userDetails.email
+				name = userDetails.name
+				enrollmentType = common.ENROLLED
+			} else {
+				userId = userTokenData.id
+				email = userTokenData.email
+				name = userTokenData.name
+				emailTemplateCode = process.env.MENTEE_SESSION_ENROLLMENT_BY_MANAGER_EMAIL_TEMPLATE // update with new template
+				enrollmentType = common.INVITED
+			}
 			const session = await sessionQueries.findById(sessionId)
```

Comment:
@VISHNUDAS-tunerlabs can we avoid finding the session details for all the mentees in case we are calling addMentees API, otherwise it will be executed if based on the mentees